core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -58,10 +60,12 @@
    * @param file the file the block belongs to
    * @param opType the OutStream's write type
    * @param blockIndex the index of the block in the file
+   * @param tachyonConf the TachyonConf instance for this file output stream.
    * @throws IOException
    */
-  BlockOutStream(TachyonFile file, WriteType opType, int blockIndex) throws IOException {
-    super(file, opType);
+  BlockOutStream(TachyonFile file, WriteType opType, int blockIndex, TachyonConf tachyonConf)
```

Comment:
How about any changes to the design is anther PR? The idea of a context has the same issue of passing it around.  The only two ways I can see that working is static state that stores the info (which is the world we area trying to move away from) and thread local which is it's own breast that makes the logic more complex.

Really don't think that it's a good idea atm.  Would love to see a simpler design using the idea, but don't think it's a good idea to see in this PR.  @uronce-cc would you be willing to take in the design for that after config goes in?